data_structures/check_binary_search_tree.py

```diff
@@ -74,85 +76,80 @@ def _display_aux(self):
         return lines, n + m + u, max(p, q) + 2, n + u // 2
 
 
-class Binary_tree:
+class BinartTree:
     def __init__(self):
         self.root = None
 
+    def add_node(self, sub_root, n1=None, n2=None):
+        sub_root.left = n1
+        sub_root.right = n2
+
+    def build_tree(self, py_list):
+        py_list_len = len(py_list)
+        q = deque()
+        # create the root
+        if py_list_len == 0:
+            return self
+
+        self.root = Node(py_list[0])
+        sub_root = self.root
+        for i in range(1, py_list_len, 2):
+            n1 = Node(py_list[i])
+            q.append(n1)
+            try:
+                n2 = Node(py_list[i + 1])
+                q.append(n2)
+            except IndexError:
+                n2 = None
+
+            self.add_node(sub_root, n1, n2)
+
+            sub_root = q.popleft()
+
+        return self
+
+    # check if binary tree is a binary search tree
+    def check_bt(self):
```

Comment:
change name to `is_binary_search_tree`.

`check_bt` doesn't make any sense. Its a BinaryTree by default. Minimally you would want to call it `check_if_bst`, but`is_binary_search_tree` is better since its explicit about what it is checking.

Reply:
member method name changed to is_binary_search_tree